README.md

```diff
@@ -1,2 +1,138 @@
+
+```elm
+type Msg
+    = ...
+    | HandleSelectUpdate ( SingleSelect.Msg Product, Maybe Product )
+```
+
+4. Configure the settings for the smart select. The `Settings` type as defined in the `SmartSelectSingle` module is below:
+
+```elm
+type alias Settings msg a =
+    { internalMsg : ( Msg a, Maybe a ) -> msg
+    , optionLabel : a -> String
+    , optionDescription : a -> String
+    , searchFn : SearchUnion a
+    , debounceDuration : Float
+    , characterSearchThreshold : Int
+    , closeOnSelect : Bool
+    }
+```
+
+- The `internalMsg` field takes a function that expects a tuple containing a SmartSelect.Msg and the selection state and returns the msg we defined in step 2.
+- `optionLabel` and `optionDescription` are functions that expect an instance of the data being selected from and return strings. Because the smart select is unaware of the type and structure of the data it is processing, these functions are necessary to help render the options in the select dropdown.
+- `debounceDuration` indicates how long if at all to wait between the last keypress and executing a search. This is particularly useful if the search being executed is pinging an external source.
+- `characterThreshold` indicates how many if any characters should be typed before a search is executed.
+- `closeOnSelect` indicates whether or not the `SmartSelect` should close itself after a selection has been made.
+
+The `searchFn` field expects a `SearchUnion`. The `SearchUnion` type as defined in the `SmartSelect.Utilities` module is below:
+
+```elm
+type SearchUnion a
+    = Local (String -> List a)
+    | API (Color.Color, ApiSearchAttrs a)
+```
+
+The `SmartSelect` can execute searches on either a `Local` or `API` basis. If the data you wish to select from has already been loaded onto the model or only exists on the frontend, use `Local`.
+It expects a function that takes in a search string and returns a list of data to be rendered. If you wish to query/search data from an external source for your selection set, use `API`. `API` expects a tuple with a `Color.Color` to indicate what color the loading spinner should be, as well as attributes to help execute the proper query. The `ApiSearchAttrs` type as defined in the `SmartSelect.Utilities` module can be seen below:
+
+```elm
+type alias ApiSearchAttrs a =
+    { headers : List Http.Header
+    , url : String -> String
+    , optionDecoder : Json.Decode.Decoder a
+    }
+```
+
+5. Viewing the select. Call `SingleSelect.view`. **_Note:_** The `view` for the multi select expects different arguments than seen here. Please refer to the `SmartSelectMulti` documentation for further information.
+
+`SingleSelect.view` expects the following (in order):
+- a boolean indicating if the select is disabled or not
+- a string indicating the name of the data being selected i.e. "Product"
+- a string indicating the currently selected entity if any i.e "Vase"
```

Comment:
What is this can this maybe be moved into the settings too?

Reply:
I should be able to move this into settings as well.